AppCenterData/AppCenterData/Internal/Util/MSDocumentUtils.m

```diff
@@ -163,15 +163,24 @@ + (MSDocumentWrapper *)documentWrapperFromDictionary:(NSDictionary *)dictionary
     deserializedValue = [(id<MSSerializableDocument>)[documentType alloc] initFromDictionary:dictionary];
     MSLogDebug([MSData logTag], @"Successfully deserialized document: %@ (partition: %@)", documentId, partition);
   }
-  return [[MSDocumentWrapper alloc] initWithDeserializedValue:deserializedValue
-                                                    jsonValue:jsonValue
-                                                    partition:partition
-                                                   documentId:documentId
-                                                         eTag:eTag
-                                              lastUpdatedDate:lastUpdatedDate
-                                             pendingOperation:pendingOperation
-                                                        error:dataError
-                                              fromDeviceCache:fromDeviceCache];
+  if (dataError) {
+    return [[MSDocumentWrapper alloc] initWithError:dataError
```

Comment:
We need a unit test for this code path.

Reply:
@dhei pair after lunch? Don't have my head around how the tests work yet.